x/pocketcore/types/pocketNode.go

```diff
@@ -0,0 +1,123 @@
+package types
+
+import (
+	"fmt"
+	"github.com/pokt-network/pocket-core/crypto"
+	"github.com/pokt-network/pocket-core/types"
+	sdk "github.com/pokt-network/pocket-core/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/privval"
+	"sync"
+)
+
+// GlobalEvidenceCache & GlobalSessionCache is used for the first pocket node and acts as backwards-compatibility for pre-lean pocket
+var GlobalEvidenceCache *CacheStorage
+var GlobalSessionCache *CacheStorage
+
+var GlobalPocketNodes = map[string]*PocketNode{}
+
+// PocketNode represents an entity in the network that is able to handle dispatches, servicing, challenges, and submit proofs/claims.
+type PocketNode struct {
+	PrivateKey      crypto.PrivateKey
+	EvidenceStore   *CacheStorage
+	SessionStore    *CacheStorage
+	DoCacheInitOnce sync.Once
+}
+
+func (n PocketNode) GetAddress() sdk.Address {
```

Comment:
Don't know if intentional but i believe this should be (n *PocketNode) because the struct is using Sync.Once and all usages come from pointer, using it this way may be copying the lock which is not recommended by the sync package.

Reply:
To my understanding, since the lock is being copied over but not being used in this method, should be fine. But I rather not have any side effects in the future anyways, so good call out. will make the change